File: app/src/main/java/com/example/website/consulta/CustomAdapter.java
package com.example.website.consulta;

import android.app.Activity;
import android.service.autofill.TextValueSanitizer;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;

public class CustomAdapter extends BaseAdapter {
    private Activity mActivity;
    private ArrayList<List> mList;
    private static LayoutInflater mInflater = null;
    private List tempValues = null;
    private int i = 0;

    public CustomAdapter(Activity a, ArrayList d) {
        mActivity = a;
        mList = d;
    }

    @Override
    public int getCount() {
        if (mList.size() <= 0)
            return 1;
        return mList.size();
    }

    @Override
    public Object getItem(int position) {
        return position;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    public static class ViewHolder {
        public TextView txtcodbar;
        public TextView txtalternante;
        public TextView txtdescripcion;
        public TextView txtsaldo;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View vi = convertView;
        ViewHolder holder;
        if (convertView == null) {
            vi = mInflater.inflate(R.layout.activity_motores, null);
            holder = new ViewHolder();
            holder.txtcodbar = vi.findViewById(R.id.txtcodbar);
            holder.txtalternante = vi.findViewById(R.id.txtalternante);
            holder.txtdescripcion = vi.findViewById(R.id.txtdescrip);
            holder.txtsaldo = vi.findViewById(R.id.txtsaldo);
            vi.setTag(holder);
        } else
            holder = (ViewHolder) vi.getTag();

        try {
            if (mList.size() <= 0) {
                holder.txtcodbar.setText("No data");
                holder.txtalternante.setText("No data");
                holder.txtdescripcion.setText("No data");
                holder.txtsaldo.setText("No data");
            }
            else
                {
                tempValues = null;
                tempValues = (List) mList.get(position);
                holder.txtcodbar.setText(tempValues.getCodbar());
                holder.txtalternante.setText(tempValues.getAlternante());
                holder.txtdescripcion.setText(tempValues.getDescripcion());
                holder.txtsaldo.setText(tempValues.getSaldo());
              }
        }
        catch(Exception ex)
        {
            ex.printStackTrace();

        }
        return null;
    }
}

File: app/src/main/java/com/example/website/consulta/View/CustomAdapter.kt
package com.example.website.consulta.View

import android.app.Activity
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import java.util.*

class CustomAdapter(private val mActivity: Activity, d: ArrayList<*>) : BaseAdapter() {
    private val mList: ArrayList<Any>? = null
    private var tempValues: Any? = null
    private val i = 0
    override fun getCount(): Int {
        if (mList != null) {
            return if (mList.size <= 0) 1 else mList.size
        }
        return 0
    }

    override fun getItem(position: Int): Any {
        return position
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    class ViewHolder {
        var txtcodbar: TextView? = null
        var txtalternante: TextView? = null
        var txtdescripcion: TextView? = null
        var txtsaldo: TextView? = null
    }

    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        var vi = convertView
        val holder: ViewHolder
        holder = vi.tag as ViewHolder
        try {
            if (mList?.size!! <= 0) {
                holder.txtcodbar!!.text = "No data"
                holder.txtalternante!!.text = "No data"
                holder.txtdescripcion!!.text = "No data"
                holder.txtsaldo!!.text = "No data"
            } else {
                tempValues = null
                tempValues = mList?.get(position)
                /*holder.txtcodbar?.setText(tempValues!!.codbar)
                holder.txtalternante?.setText(tempValues!!.alternante)
                holder.txtdescripcion?.setText(tempValues!!.descripcion)
                holder.txtsaldo?.setText(tempValues!!.saldo)*/
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
        return vi
    }
}